src/http/client.rs

```diff
@@ -34,41 +32,32 @@ impl Client {
                     }
                 }
             },
-            Err(_) => {
+            Err(e) => {
+                println!("Error connecting upstream: {}", e);
                 downstream.write_all(b"HTTP/1.1 501 Internal Server Error\r\nContent-Length: 6\r\n\r\nSorry\n").unwrap();
             }
         }
     }
 
     pub fn connect(&self, url: &url::Url) -> io::Result<mioco::tcp::TcpStream> {
-        let domain = url.host_str().unwrap();
-        let port = match url.port() {
-            Some(port) => port,
-            None => {
-                match url.scheme() {
-                    "http" => 80,
-                    "https" => 443,
-                    _ => panic!("Unknown scheme {}", url.scheme())
-                }
-            }
-        };
+        use std::net::ToSocketAddrs;
```

Comment:
It can be! It turns out you can do them inline which I found interesting. I'll resubmit a bit later with it moved to the top.